### PR TITLE
SDFSOF-63 - Added implementation of endpoint for handling RPC calls with corresponding request validation and basic error handling

### PR DESCRIPTION
**Pull request by MazurakIhor**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Implementation of endpoint for handling RPC calls with request validation and error handling

### Why

This endpoint will handle RPC calls that should replace PATCH /transactions REST endpoint

### CODE REVIEW

**Comment by MazurakIhor** [comment]
softserve_develop branch is 17 commits behind develop but changes are small and easy to merge.
I am going to merge latest develop into softserve_develop branch after this PR is merged, just to not mix my changes with changes from develop branch. But if you think it's better to merge latest develop right now - NP, I will merge it.

**Comment by JakeUrban** [comment]
Either approach is fine by me, whatever you think is best.

File: api-schema/src/main/java/org/stellar/anchor/api/rpc/RpcRequest.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.stellar.anchor.api.rpc;
+
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class RpcRequest {
+  private Object id;
```
Comment (Ifropc) [inline api-schema/src/main/java/org/stellar/anchor/api/rpc/RpcRequest.java:9]: Pretty sure it's just String

> id
    An identifier established by the Client that MUST contain a String, Number, or NULL value if included. If it is not included it is assumed to be a notification. The value SHOULD normally not be Null [[1]](https://www.jsonrpc.org/specification#id1) and Numbers SHOULD NOT contain fractional parts [[2]](https://www.jsonrpc.org/specification#id2)

(from specs)

File: api-schema/src/main/java/org/stellar/anchor/api/rpc/RpcRequest.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  private String jsonrpc;
+  private String method;
+  private Object params;
```
Comment (Ifropc) [inline api-schema/src/main/java/org/stellar/anchor/api/rpc/RpcRequest.java:12]: I'm guessing it's not possible to make it generic, on what step this going to be converted to an actual object?

**Review by Ifropc — APPROVED** [review]

File: api-schema/src/main/java/org/stellar/anchor/api/exception/RpcException.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+@Getter
+@EqualsAndHashCode(callSuper = false)
+public class RpcException extends AnchorException {
+  private final int errorCode;
+  private Object additionalData;
+
+  public RpcException(int errorCode, String message) {
```
Comment (Ifropc) [inline api-schema/src/main/java/org/stellar/anchor/api/exception/RpcException.java:12]: Should it has RpcErrorCode as parameter instead of int?

Comment (MazurakIhor) [reply api-schema/src/main/java/org/stellar/anchor/api/exception/RpcException.java]: Changed `int errorCode` to `RpcErrorCode errorCode`

File: api-schema/src/main/java/org/stellar/anchor/api/rpc/RpcRequest.java (hunk 1 above)
Comment (MazurakIhor) [reply api-schema/src/main/java/org/stellar/anchor/api/rpc/RpcRequest.java]: Request may contain `"id": "1"` or `"id": 1`.
String will handle both cases and convert to String with value "1"
Response will contain String as well, even if request contained Number. But it's shoun't be a problem to convert it back to Number.

Comment (Ifropc) [reply api-schema/src/main/java/org/stellar/anchor/api/rpc/RpcRequest.java]: Oh yeah you are right gson treats it differently. Let's keep it like this then

**Pull request merged**